homeassistant/components/thread/dataset_store.py

```diff
@@ -310,6 +325,65 @@ def preferred_dataset(self, dataset_id: str) -> None:
         self._preferred_dataset = dataset_id
         self.async_schedule_save()
 
+    async def _set_preferred_dataset_if_only_network(
+        self, dataset_id: str, border_agent_id: str
+    ) -> None:
+        """Set the preferred dataset, unless there are other routers present."""
+        _LOGGER.debug(
+            "_set_preferred_dataset_if_only_network called for router %s",
+            border_agent_id,
+        )
+
+        own_router_evt = Event()
+        other_router_evt = Event()
+
+        @callback
+        def router_discovered(
+            key: str, data: discovery.ThreadRouterDiscoveryData
+        ) -> None:
+            """Handle router discovered."""
+            _LOGGER.debug("discovered router with id %s", data.border_agent_id)
+            if not data.border_agent_id:
+                return
+
+            if data.border_agent_id == border_agent_id:
+                own_router_evt.set()
+                return
```

Comment:
It seems that at least Apple BR do not announce a border agent id currently. Yet we should not make our BR the default in this cases.

So I'd suggest to consider BRs without border agent id as "other" as well:

```suggestion
            if data.border_agent_id and data.border_agent_id == border_agent_id:
                own_router_evt.set()
                return
```

Reply:
OK, let's do it 👍